pyteal/compiler/sort.py

```diff
@@ -13,22 +14,30 @@ def sortBlocks(start: TealBlock) -> List[TealBlock]:
         An ordered list of TealBlocks that is sorted such that every block is guaranteed to appear
         in the list before all of its outgoing blocks.
     """
-    # based on Kahn's algorithm from https://en.wikipedia.org/wiki/Topological_sorting
     S = [start]
     order = []
-
+    visited = set()  # I changed visited to a set to be more efficient
     while len(S) != 0:
-        n = S.pop(0)
+        n = S.pop()
+
+        if id(n) in visited:
+            continue
+
+        S += n.getOutgoing()
+
         order.append(n)
-        for i, m in enumerate(n.getOutgoing()):
-            for i, block in enumerate(m.incoming):
-                if n is block:
-                    m.incoming.pop(i)
-                    break
-            if len(m.incoming) == 0:
-                if i == 0:
-                    S.insert(0, m)
-                else:
-                    S.append(m)
+        visited.add(id(n))
+
+    endIndex = -1
+    for i, block in enumerate(order):
+        if block == end:
```

Comment:
Can you check if `block is end` instead? This is a slightly stronger check, since we want to make sure we find the end block exactly, not a different block which happens to have the same ops as the end block, which is what `==` checks.